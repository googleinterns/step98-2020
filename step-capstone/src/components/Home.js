import React from 'react';
import {Button, createMuiTheme, Grid, responsiveFontSizes, Typography, ThemeProvider, withStyles } from '@material-ui/core'
import SignInWidget from './Firebase/SignInWidget'
import '../styles/Home.css'
import logo from '../images/logo.png'
import tickets from '../images/travel-tickets-colour-thumbnail.png'
import calendar from '../images/calendar-colour-thumbnail.png'
import help from '../images/drawkit-support-woman-colour-thumbnail.png'
import map from '../images/map-colour-thumbnail.png'

const LoginButton = withStyles({
  root: {
    background: 'white',
    borderRadius: 3,
    border: 0,
    color: 'black',
    height: 70,
    width: 250,
    padding: '0 30px',
    boxShadow: '0 3px 5px 2px rgba(0, 0, 0, .3)',
  }, 
  label: {
    fontSize: '20px'
  }
})(Button);

class Home extends React.Component {
  constructor(props){
    super(props);
    this.state = {clickedToLogin : false};
    this.onLoginClick = this.onLoginClick.bind(this);
  }

  onLoginClick() {
    this.setState({clickedToLogin : true});
  }


  render() {
    let theme = createMuiTheme();
    theme = responsiveFontSizes(theme);

    return (
      <ThemeProvider theme={theme}>
        <Grid container id='home'>
          <Grid item id='top-half'>
            <Typography id='banner-header' variant='h2'> Less planning. More exploring. </Typography> 
            <img id='logo' src={logo} alt='ZED logo'/>
            
            {this.state.clickedToLogin ?  <div id='login-firebase'><SignInWidget/></div> :
                <LoginButton id='login-button'
                variant='contained' 
                onClick={this.onLoginClick}>
                Get Started   
                </LoginButton>
            }
          </Grid>

          <Grid item id='bottom-half'>
            <br/>
            <Typography variant='h2'> Features </Typography>
            <br/>
            <Grid container id='features' wrap='nowrap' direction='row' justify='space-around'>
              <Grid item>
                <img id='tickets-icon' src = {tickets} alt = 'travel tickets icon' height='250px' width='250px'  />
                <br/>
                <Typography variant='subtitle-1'> Add flights, hotels, restaurants, 
                concerts, sightseeing spots, and anything else you want to see or do. 
                </Typography>
              </Grid>
              <Grid item >
                <img id='calendar-icon' src = {calendar} alt = 'calendar icon' height='250px' width='250px'/>
                <br/>
                <Typography variant='subtitle-1'> Take the stress out of scheduling by 
                using our calendar to keep track of where you're going and when. 
                </Typography>
              </Grid>
              <Grid item >
                <img id='map-icon' src = {map} alt = 'map icon' height='250px' width='250px' />
                <br/>
                <Typography variant='subtitle-1'> Trying to optimize your day? Use our map 
                to plot your perfect route - or let us do it for you. 
                </Typography>
              </Grid>
              <Grid item >
                <img id='help-icon' src = {help} alt = 'help icon' height='250px' width='250px' />
                <br/>
                <Typography variant='subtitle-1'> Running out of ideas? We can give you 
                suggestions tailored to your schedule and preferences. 
                </Typography>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </ThemeProvider>
    )
  }
}

export default Home;