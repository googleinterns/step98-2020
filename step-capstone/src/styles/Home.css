#top-half{
  height:70%;
  width:100%;
  background-image:url('../images/hot-air-ballons-in-the-sky2.jpg');
  background-size: cover;
  background-repeat: no-repeat ;
  position: relative;
}
#bottom-half{
  background-color: white;
  height:30%;
  width:100%;
  text-align: center;
}
#home{
  direction: column;
  height:100%;
  width:100%;
}
#login-button{
  top: 80%;
  left:50% ;
  position: absolute;
  transform: translate(-50%, -50%);
} 
#login-firebase{
  position:absolute;
  bottom:5%;
  left:50%;
  transform: translate(-50%, 0);
} 
#banner-header{
  top:50%;
  left:50%;
  color: white;
  font-weight: bold;
  position: absolute;
  transform: translate(-50%, -50%);
}
#features{
  height: 100%;
  width:100%;
}
