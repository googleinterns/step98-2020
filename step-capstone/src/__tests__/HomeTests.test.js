import React from 'react';
import Home from '../components/Home'
import renderer from 'react-test-renderer';

test('Home page displays correctly', () => {
  const homeComponent = renderer.create(
    <Home />,
  );
  let test = homeComponent.toJSON();
  expect(test).toMatchSnapshot();

});