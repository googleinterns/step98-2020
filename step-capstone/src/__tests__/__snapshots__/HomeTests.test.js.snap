// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page displays correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
  id="home"
>
  <div
    className="MuiGrid-root MuiGrid-item"
    id="top-half"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
      id="banner-header"
    >
       Less planning. More exploring. 
    </h2>
    <img
      alt="ZED logo"
      id="logo"
      src="logo.png"
    />
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-contained"
      disabled={false}
      id="login-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label WithStyles(ForwardRef(Button))-label-2"
      >
        Get Started
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
    id="bottom-half"
  >
    <br />
    <h2
      className="MuiTypography-root MuiTypography-h2"
    >
       Features 
    </h2>
    <br />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-around"
      id="features"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="travel tickets icon"
          height="250px"
          id="tickets-icon"
          src="travel-tickets-colour-thumbnail.png"
          width="250px"
        />
        <br />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
           Add flights, hotels, restaurants, concerts, sightseeing spots, and anything else you want to see or do.
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="calendar icon"
          height="250px"
          id="calendar-icon"
          src="calendar-colour-thumbnail.png"
          width="250px"
        />
        <br />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
           Take the stress out of scheduling by using our calendar to keep track of where you're going and when.
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="map icon"
          height="250px"
          id="map-icon"
          src="map-colour-thumbnail.png"
          width="250px"
        />
        <br />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
           Trying to optimize your day? Use our map to plot your perfect route - or let us do it for you.
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="help icon"
          height="250px"
          id="help-icon"
          src="drawkit-support-woman-colour-thumbnail.png"
          width="250px"
        />
        <br />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
        >
           Running out of ideas? We can give you suggestions tailored to your schedule and preferences.
        </h6>
      </div>
    </div>
  </div>
</div>
`;
